### DISCUSSION TITLE
Problema cuando le meto estacion de origen

### DISCUSSION THREAD

**Discussion by itzyeraycano** [emartinez-dev/renfe-bot#4] [Q&A] [answered]
Cuando le pongo al bot de telegram la estacion de origen, no me contesta y en la terminal me da varios problemas, espero que me podais ayudar, adjunto fotos
![image](https://github.com/emartinez-dev/renfe-bot/assets/123980378/78fae39b-b284-443d-9a9a-f18fd2ae6ff4)
![image](https://github.com/emartinez-dev/renfe-bot/assets/123980378/a7098fab-e10c-413e-aa08-0a550c0ad2ad)

**Comment by emartinez-dev** [comment]
Hola! Claro, con mucho gusto.
Parece que se trata de un problema con las rutas de los archivos, además veo que estás utilizando Windows. Intenta lo siguiente:

1. Abre cmd y navega hasta la carpeta en la que tienes el bot, en este caso lo podrías hacer con `cd C:\Users\jerry\Downloads\renfe-bot-master\`
2. Asegurate de tener instaladas las dependencias del programa, ejecuta `python -m pip install -r requirements.txt`
3. Una vez termine de instalarse, ejecuta el bot poniendo `python bot/renfebot.py` en la terminal.

Dime si te ha servido e intentaré ayudarte a solucionarlo.

**Reply by itzyeraycano** [reply]
Sigue sin funcionarme, me aparece este error

![image](https://github.com/emartinez-dev/renfe-bot/assets/123980378/572923d7-4b22-4c30-8f68-bdb96c60be8f)

**Reply by emartinez-dev** [reply]
Puede tratarse de un problema de encoding del archivo .txt, ya que yo creé el archivo en Linux y lo estás ejecutando en Windows, prueba a cambiar esta línea:

https://github.com/emartinez-dev/renfe-bot/blob/993220fda79a4f0d1034fe1ffa1c45b303eec0cb/bot/utils.py#L7

Por esta otra:

```python
with open('resources/stations.txt', 'r', encoding='utf-8') as f:
```

Si aún así no te funciona, puedes renombrar el archivo resources/stations.txt, copiar el contenido y crear un .txt nuevo con el mismo nombre y contenido, eso debería funcionar.

**Reply by itzyeraycano** [reply]
El bot me corrio pero ahora me da un problema a la hora de encontrar los billetes 
![image](https://github.com/emartinez-dev/renfe-bot/assets/123980378/887687e8-690d-4727-9ef5-99c1f043467c)
![image](https://github.com/emartinez-dev/renfe-bot/assets/123980378/6d416b95-afe7-47d4-acef-8f264b20b85f)

**Reply by emartinez-dev** [reply]
Puedes enviarme el log más reciente de la carpeta `logs/`?

**Reply by itzyeraycano** [reply]
[scraper_15_08_10_10_2023.log](https://github.com/emartinez-dev/renfe-bot/files/12857647/scraper_15_08_10_10_2023.log)

**Reply by emartinez-dev** [reply]
Me aparece vacío al descargarlo, prueba a abrirlo y pegar el contenido aquí

**Comment by itzyeraycano** [comment]
Todos los logs estan vacios

**Reply by emartinez-dev** [reply]
Vaya, que faena. Se abre el Chrome al menos y puedes ver cómo intenta buscar los billetes?

**Comment by itzyeraycano** [comment]
Hace por abrirse una pestaña pero se cierra y da el error que te he mandado
antes

**Reply by emartinez-dev** [reply]
He conseguido reproducir un error, pero el que me estabas pasando tiene la traza de error cortada, intenta pegar el texto completo. 

Aquí está la mía, y parece que es porque mi Google Chrome tiene una versión distinta que Chromedriver (el programa que se encarga de manejar el navegador), ¿te aparece a ti también este error?

```bash
Traceback (most recent call last):
  File "/home/emartinez/renfe-bot/bot/renfebot.py", line 211, in search_trains
    watcher = Watcher(watcher_params["origin_station"], watcher_params["destination_station"], \
  File "/home/emartinez/renfe-bot/bot/watcher.py", line 17, in __init__
    self.scraper = RenfeScraper()
  [...]
```

> selenium.common.exceptions.SessionNotCreatedException: Message: session not created: **This version of ChromeDriver only supports Chrome version 114
> Current browser version is 117.0.5938.92 with binary path /usr/bin/google-chrome**

```bash
[...]
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/emartinez/renfe-bot/bot/renfebot.py", line 231, in <module>
    bot.polling()
  File "/home/emartinez/renfe-bot/.venv/lib/python3.10/site-packages/telebot/__init__.py", line 1043, in polling
    self.__threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "/home/emartinez/renfe-bot/.venv/lib/python3.10/site-packages/telebot/__init__.py", line 1118, in __threaded_polling
    raise e
  File "/home/emartinez/renfe-bot/.venv/lib/python3.10/site-packages/telebot/__init__.py", line 1074, in __threaded_polling
    self.worker_pool.raise_exceptions()
  File "/home/emartinez/renfe-bot/.venv/lib/python3.10/site-packages/telebot/util.py", line 147, in raise_exceptions
    raise self.exception_info
  File "/home/emartinez/renfe-bot/.venv/lib/python3.10/site-packages/telebot/util.py", line 90, in run
    task(*args, **kwargs)
  File "/home/emartinez/renfe-bot/bot/renfebot.py", line 125, in get_filter
    search_trains(message, user_params)
  File "/home/emartinez/renfe-bot/bot/renfebot.py", line 217, in search_trains
    watcher.driver.quit()
UnboundLocalError: local variable 'watcher' referenced before assignment
```

**Comment by itzyeraycano** [comment]
[image: image.png]

**Comment by miguelgonzalez-27** [comment]
Me ocurre lo mismo, no consigo hacerlo funcionar. Sigue funcionando o está desactualizado?

**Comment by emartinez-dev** [comment, marked as answer]
Hola @itzyeraycano @xxboostxx 

He actualizado el bot para que utilice otro mecanismo a la hora de buscar los billetes, he estado utilizándolo un par de meses y me ha funcionado perfectamente en Ubuntu / Debian, no tengo un equipo con Windows para poder probarlo pero debería funcionar.

Seguid las nuevas instrucciones, instalad todas las dependencias del `requirements.txt` y también las de playwright con `playwright install` y `playwright install-deps`. 

Si no os funciona, también tenéis una imagen de Docker que recopila todas las dependencias y corre el código en un container. Voy a cerrar el tema, si os encontráis algún bug a la hora de instalarlo o usándolo, podéis crear un issue para que lo investigue.

Felices fiestas!
